.github/workflows/release.yaml

```diff
@@ -35,7 +38,11 @@ jobs:
         uses: fluxcd/flux2/action@v2.2.2
       - name: Clone repo
         uses: actions/checkout@v4
-
+      - name: Create Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh release create --draft --generate-notes "${{ github.ref_name }}"
```

Comment:
Our current way of working for creating a release is that we do it manually through the GitHub UI. Currently not too keen on changing that right now.
My understanding of the `gh release upload` is that it will be able to find the release from the tag.

```shell
➜ gh release upload --help
Upload asset files to a GitHub Release.

To define a display label for an asset, append text starting with %#% after the
file name.

USAGE
  gh release upload <tag> <files>... [flags]
```

Do I understand it correct?

Reply:
yes it will work if the release is there befor the CI/CD reached the upload step.

if you create the release before the tag in git everything will work.
if the tag is created before the relase you may run in to problems, but best is to test it.
in the worst case the CI/CD needs to be restarted and at that point it can be adapted for the next release.